daemon/container_unix.go

```diff
@@ -914,6 +914,12 @@ func (container *Container) allocateNetwork() error {
 		if mode.IsDefault() {
 			networkName = controller.Config().Daemon.DefaultNetwork
 		}
+		if mode.IsUserDefined() {
+			if n, err := container.daemon.FindNetwork(networkName); err == nil {
+				networkName = n.Name()
```

Comment:
It is not this function's concern to do that validation. This function only cares of getting an hold of the real name for a user defined network, in case the network was passed by id. It is a best effort.

The function which will fail (in the situation you described) is the one effectively dealing with the network controller, later on: 
`func (container *Container) connectToNetwork()`

We are not modifying that logic with this change.